src/utils/zod.ts

```diff
@@ -106,6 +106,18 @@ export type ZodTypeCoercible = z.ZodNumber | z.ZodBoolean | z.ZodBigInt | z.ZodD
 
 export const instanceofZodTypeCoercible = (_type: z.ZodTypeAny): _type is ZodTypeCoercible => {
   const type = unwrapZodType(_type, false);
+
+  if (instanceofZodTypeKind(type, z.ZodFirstPartyTypeKind.ZodUnion)) {
+    return type._def.options.every(
+      (option) => instanceofZodTypeLikeString(option) || instanceofZodTypeCoercible(option),
+    );
+  }
```

Comment:
A value is coercible if it itself is coercible or if it is a union of only coercible values.